spec/controllers/categories_controller_spec.rb

```diff
@@ -1,187 +1,204 @@
-      it { is_expected.not_to set_flash }
     end
-    context 'not admin' do
-      it 'should redirect to root_url' do
-        sign_in FactoryGirl.create(:user)
-        get :edit, id: @category
-        expect(response).to redirect_to(root_url)
+    context 'user is not admin' do
+      before do
+        mock_user_sign_in
+        put :update, id: 1, category: { id: 2 }
       end
+      it_behaves_like 'redirected request'
     end
   end
-  describe 'PUT update' do
+
+  describe 'PUT deactivate' do
     context 'is admin' do
-      before(:each) do
-        sign_in FactoryGirl.create(:admin)
-      end
-      context 'with valid attributes' do
-        before(:each) do
-          put :update,
-              id: @category,
-              category: FactoryGirl.attributes_for(:category, name: 'Updated')
+      before { mock_user_sign_in(mock_user(:admin)) }
+      shared_examples 'not confirmed' do |flash_type, **opts|
+        let!(:cat) { FactoryGirl.build_stubbed(:category) }
+        before do
+          allow(Category).to receive(:find).with(cat.id.to_s).and_return(cat)
+          allow(cat).to receive(:destroy)
+          put :deactivate, id: cat.id, **opts
+        end
+        it { is_expected.to set_flash[flash_type] }
+        it { is_expected.to redirect_to(cat) }
+        it "doesn't destroy the category" do
+          expect(cat).not_to have_received(:destroy)
         end
-        it 'should set @category to the correct category' do
-          expect(assigns(:category)).to eq(@category)
+      end
+      it_behaves_like 'not confirmed', :notice, deactivation_cancelled: true
+      it_behaves_like 'not confirmed', :error
+
+      context 'confirmed' do
+        let!(:cat) do
+          CategoryMock.new(traits: [:findable], equipment_models: [])
         end
-        it 'should successfully save new attributes to the database' do
-          @category.reload
-          expect(@category.name).to eq('Updated')
+        before do
+          request.env['HTTP_REFERER'] = 'where_i_came_from'
+          put :deactivate, id: cat.id, deactivation_confirmed: true
         end
-        it { is_expected.to redirect_to(@category) }
-        it { is_expected.to set_flash }
-      end
-      context 'with invalid attributes' do
-        before(:each) do
-          put :update,
-              id: @category,
-              category: FactoryGirl.attributes_for(:category,
-                                                   name: nil,
-                                                   max_per_user: 10)
+        it 'destroys the category' do
+          expect(cat).to have_received(:destroy)
         end
-        it 'should not update attributes of @category in the database' do
-          @category.reload
-          expect(@category.name).not_to be_nil
-          expect(@category.max_per_user).not_to eq(10)
+      end
+
+      context 'with reservations' do
+        let!(:cat) { CategoryMock.new(traits: [:findable]) }
```

Comment:
Could you clarify on why you chose to use a CategoryMock in this destroy case and built a stubbed category above? Thanks.